flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/matrix/BLAS.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.matrix;
+
+/**
+ * A utility class that wraps netlib BLAS and provides some operations on dense matrix
+ * and dense vector.
+ */
+public class BLAS {
+	private static final com.github.fommil.netlib.BLAS BLAS = com.github.fommil.netlib.BLAS.getInstance();
+
+	/**
+	 * y += a * x .
+	 */
+	public static void axpy(double a, double[] x, double[] y) {
+		BLAS.daxpy(x.length, a, x, 1, y, 1);
+	}
+
+	public static void axpy(double a, DenseVector x, DenseVector y) {
+		axpy(a, x.getData(), y.getData());
+	}
+
+	/**
+	 * x \cdot y .
+	 */
+	public static double dot(double[] x, double[] y) {
+		return BLAS.ddot(x.length, x, 1, y, 1);
+	}
+
+	public static double dot(DenseVector x, DenseVector y) {
```

Comment:
missing javadoc? this also prompts me to suggest wrapper on `Vector` and make all these concrete impl private.

Reply:
Blas is a package that provides high performance linear algebra operations, and it is not intended to work on sparse vector and sparse matrix cases. So it is better to be what it is now.